### DISCUSSION TITLE
v1.122.2 - build b

### DISCUSSION THREAD

**Discussion by 4v3ngR** [4v3ngR/immich-native-macos#28] [Announcements]
Alpha software.

The second build for v1.122.2 adds `/opt/homebrew/bin` to the PATH variable. This should fix installations on apple silicon 🤞 

<hr /><em>This discussion was created from the release <a href='https://github.com/4v3ngR/immich-native-macos/releases/tag/v1.122.2-b'>v1.122.2 - build b</a>.</em>

**Comment by patchuby** [comment]
Hello, thank you for your work.

I've run the installer but it says the installation was successful while the immich-install.log last line is `INFO: install daemons`.
Trying to access 127.0.0.1:3001(or other ports) doesn't seem to work. Am I missing something ?

**Comment by 4v3ngR** [comment]
In the "activity monitor" app, do you see any immich processes running?

![Screenshot 2024-12-11 at 6 47 07 am](https://github.com/user-attachments/assets/b90a24f9-3e20-49f2-9d0d-809fe518a552)

**Reply by aerlenbach** [reply]
I have the same problem. Also tried to run `./build.sh` (per the instructions) in terminal, resulting in `zsh: no such file or directory: ./build.sh`. 

Activity Monitor shows immich is not running.

**Reply by patchuby** [reply]
> I have the same problem. Also tried to run `./build.sh` (per the instructions) in terminal, resulting in `zsh: no such file or directory: ./build.sh`.
> 
> Activity Monitor shows immich is not running.

Are you running it within the context of the directory ? You can right-click on the folder and use `New Terminal at Folder`

**Reply by aerlenbach** [reply]
> You can right-click on the folder and use `New Terminal at Folder`

I'm afraid I don't see that as an option.

When I run the installer, the Console App opens up, does its thing, then ends with:

`2024-12-10 21:23:36 (6.32 MB/s) - ‘cities500.zip’ saved [11991891/11991891]`

`Archive:  cities500.zip`

`inflating: cities500.txt`           

`INFO:  install daemons`

The installer app claims it's complete. I don't see any indication of where to run `./build.sh`

I don't install applications this way very often so there might be some obvious unwritten thing in the instructions that I'm not getting. 

edit:

[this user](https://github.com/4v3ngR/immich-native-macos/discussions/8#discussioncomment-10259284) appeared to have the same problem I did and assumes the app isn't capable of running on M series Macs.

**Reply by 4v3ngR** [reply]
I've done some searching and it appears that if the /Library/LaunchDaemons directory doesn't exist, an installer is allowed to create it. I've done a 3rd build for v1.122.3. Can you see if this works for you:

https://github.com/4v3ngR/immich-native-macos/releases/tag/v1.122.3-c

**Comment by patchuby** [comment]
No Immich in the activity monitor for me either

**Comment by patchuby** [comment]
Ran the build.sh script, managed to get to the same point after building the installer but with extra messages : 
`` INFO:  install daemons
Load failed: 5: Input/output error
Try running launchctl bootstrap as root for richer errors.``

What are the options for these commands (`domain-target`, `service-path`, etc...) , maybe I can get more useful errors ?

**Reply by 4v3ngR** [reply]
🤔 Could you do me a favour (as I don't have an M series mac)? 

Could you open the Terminal app and run the following commands? (these are the same as the installdaemons script run:

```
sudo launchctl unload -w /Library/LaunchDaemons/com.immich.plist
sudo launchctl unload -w /Library/LaunchDaemons/com.immich.machine.learning.plist

sudo launchctl load -w /Library/LaunchDaemons/com.immich.plist
sudo launchctl load -w /Library/LaunchDaemons/com.immich.machine.learning.plist
```

The first two may produce errors (that's OK). I'm interested in any output from the second 2.

**Reply by patchuby** [reply]
```
tomallibert@PFSMacMini ~ % sudo launchctl unload -w /Library/LaunchDaemons/com.immich.plist
Password:
Unload failed: 5: Input/output error
Try running `launchctl bootout` as root for richer errors.
user@PFSMacMini ~ % sudo launchctl unload -w /Library/LaunchDaemons/com.immich.machine.learning.plist
Unload failed: 5: Input/output error
Try running `launchctl bootout` as root for richer errors.
user@PFSMacMini ~ % sudo launchctl load -w /Library/LaunchDaemons/com.immich.plist
Load failed: 5: Input/output error
Try running `launchctl bootstrap` as root for richer errors.
user@PFSMacMini ~ % sudo launchctl load -w /Library/LaunchDaemons/com.immich.machine.learning.plist
Load failed: 5: Input/output error
Try running `launchctl bootstrap` as root for richer errors.
```
I must sayI  don't have any LaunchDaemons folder in Library, only a folder called LaunchAgents

**Reply by patchuby** [reply]
Ah sorry, I'm not very familiar with MacOS, I was looking at the user Library, not the root Library folder, this one has indead the folder and the referenced files

**Reply by 4v3ngR** [reply]
Quick question... what version of macOS are you using?

It could be that /Library/LaunchDaemons is deprecated, and as such is no longer created as part of the macOS installation.

I'll continue to investigate it here. I'll also see if the installation works by installing the plist files in LaunchAgents.

**Reply by patchuby** [reply]
I'm on Sonoma.
To be clear ~/Library (user's Library) doesn't have LaunchDaemons but /Library (root Library) has it.

**Comment by patchuby** [comment]
Also tried building using v. 122.1, no dice

**Comment by harsharora001** [comment]
It suggests to run with bootstrap for better error, and when I did it, it showed below

> launchctl bootstrap load -w /Library/LaunchDaemons/com.immich.plist
Unrecognized target specifier, did you mean
system/com.apple.iomfb_fdr_loader
system/com.apple.logkextloadsd
system/com.apple.mDNSResponderHelper.reloaded
system/com.apple.mDNSResponder.reloaded
gui/501/com.apple.securityuploadd
gui/501/io.sideloadly.daemon
gui/501/com.apple.storedownloadd
Usage: launchctl bootstrap <domain-target> [service-path, service-path2, ...]
<service-target> takes a form of <domain-target>/<service-id>.
Please refer to `man launchctl` for explanation of the <domain-target> specifiers.
